Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -18,72 +28,232 @@ class commsWithAttitude : public pathManagerState
+        static _LandingTakeoffOutput output;
+        static _WaypointManager_Data_In waypointInput;
+        static _WaypointManager_Data_Out waypointOutput;
+        takeoffClimbStage() {}
+        takeoffClimbStage(const takeoffClimbStage& other);
+        takeoffClimbStage& operator =(const takeoffClimbStage& other);
+};
+
+
 class cruisingState : public pathManagerState
 {
     public:
         void enter(pathManager* pathMgr) {(void) pathMgr;}
         void execute(pathManager* pathMgr);
         void exit(pathManager* pathMgr) {(void) pathMgr;}
         static pathManagerState& getInstance();
-        static _PathData *GetWaypointData(void) {return &_waypointdata;}
-        static _WaypointManager_Data_Out *GetOutputData(void) {return &_outputdata;}
+        static _WaypointManager_Data_Out* GetOutputData(void) {return &_outputdata;}
+        static _CruisingState_Telemetry_Return* GetErrorCodes(void) {return &_returnToGround;}
+
+        #ifdef UNIT_TESTING
+            WaypointManager* GetWaypointManager(void) {return &cruisingStateManager;}
+            int* GetWaypointIdArray(void) {return waypointIDArray;}
+        #endif
     private:
         cruisingState() {}
         cruisingState(const cruisingState& other);
         cruisingState& operator =(const cruisingState& other);
-        //WaypointManager(0,0) {} initializing instance of WaypointManager class
-        int waypointIDArray[PATH_BUFFER_SIZE]; 
-        static _PathData _waypointdata; 
+
+        WaypointManager cruisingStateManager;
+        int waypointIDArray[PATH_BUFFER_SIZE]; // Stores ids of the waypoints in the flight path in the order that they are executed
+        static _WaypointManager_Data_In _inputdata;
         static _WaypointManager_Data_Out _outputdata; 
+        static _CruisingState_Telemetry_Return _returnToGround;
+        bool inHold = false;
+        bool goingHome = false;         
+};
+
+class landingTransitionStage : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        
+        static _LandingTakeoffOutput* getControlOutput(){return &output;}
+        static WaypointManager landingPath; 
+        static _WaypointStatus waypointStatus; //used to catch errors
+        static _LandingPath path; //used to load in path
+    private:
+        int waypointIDArray[PATH_BUFFER_SIZE];
+        static _LandingTakeoffInput input;
+        static _LandingTakeoffOutput output;
+        static _WaypointManager_Data_In waypointInput;
+        static _WaypointManager_Data_Out waypointOutput;
+        static _PathData * pathArray[3]; //used to translate loaded in path to something the waypoint manager can take as a parameter
+        static _PathData * currentLocation;
+        double differenceInHeading1;
+        double differenceInHeading2;
```

Comment:
Another thing actually, I don't think we even need these to be member variables for this class. 

Instead, you can declare them in the ::execute() method for this class. That way we aren't populating this class structure with stuff that aren't really being cached or accessed elsewhere.

Reply:
true, I'll make those